### ISSUE TITLE
operator (`Linear` and `BatchNorm1d`) fusion fails in `torch.ao.quantization.fuse_modules`

### ISSUE THREAD

**Issue by AnonymousChaseDreamer** [pytorch/pytorch#112945]
### 🐛 Describe the bug

This is similar to the previous two bugs #112579 and #112820, except this time it's another operator (`Liner`) with `BatchNorm`

Here is the code to reproduce the bug
```python
from torch.ao.quantization.fuse_modules import fuse_modules
import torch.nn as nn


class M(nn.Module):
    def __init__(self):
        super(M, self).__init__()
        self.fc1 = nn.Linear(3 * 224 * 224, 64)  # Linear layer with appropriate input and output dimensions
        self.bn1 = nn.BatchNorm1d(64)  # Use BatchNorm1d for 1D input
        self.fc2 = nn.Linear(64, 64)  # Linear layer with appropriate input and output dimensions
        self.bn2 = nn.BatchNorm1d(64)  # Use BatchNorm1d for 1D input

    def forward(self, x):
        x = self.fc1(x)
        x = self.bn1(x)
        x = self.fc2(x)
        x = self.bn2(x)
        return x


m = M().eval()

modules_to_fuse = ['fc1', 'bn1']

print(m)
fused_m = fuse_modules(m, modules_to_fuse)
print(fused_m)
```

Here is the output:
```
M(
  (fc1): Linear(in_features=150528, out_features=64, bias=True)
  (bn1): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
  (fc2): Linear(in_features=64, out_features=64, bias=True)
  (bn2): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
)
M(
  (fc1): Linear(in_features=150528, out_features=64, bias=True)
  (bn1): Identity()
  (fc2): Linear(in_features=64, out_features=64, bias=True)
  (bn2): BatchNorm1d(64, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
)
```

### Versions

Collecting environment information...
PyTorch version: 2.1.0
Is debug build: False
CUDA used to build PyTorch: Could not collect
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.8.2
Libc version: glibc-2.27

Python version: 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-4.15.0-213-generic-x86_64-with-glibc2.27
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration:
GPU 0: Tesla V100-SXM2-32GB
GPU 1: Tesla V100-SXM2-32GB
GPU 2: Tesla V100-SXM2-32GB
GPU 3: Tesla V100-SXM2-32GB

Nvidia driver version: 470.182.03
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.24.3
[pip3] onnx==1.13.1
[pip3] onnxoptimizer==0.3.13
[pip3] onnxruntime==1.16.1
[pip3] torch==2.1.0
[pip3] torchaudio==2.1.0
[pip3] torchvision==0.16.0
[conda] Could not collect

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel

**Comment by jerryzh168** [comment]
replied in https://github.com/pytorch/pytorch/issues/112579
